### DISCUSSION TITLE
passing an array to function

### DISCUSSION THREAD

**Discussion by Freem11** [supabase/supabase#13304] [Questions] [answered]
Im trying to pass an array (called animals) to my function and am not sure how to declare it 
doing it as i have below gives me : Failed to run sql query: type animals[] does not exist

ultimately i want to do is loop over the array and pass its values as a variable length condition on my where clause:
for example if the array were: ['monkey', 'bear' ,'dog']
I want to add "and label='monkey' or label='bear or label='dog
but if it were: ['monkey', 'bear' ,'dog', 'cow']
it would be "and label='monkey' or label='bear or label='dog or label='cow'

can anyone guide me as to how to accomplish this? my current function is setup as below and works fine if i dont have the animals array listed: 

create or replace function dididi(max_lat integer, min_lat integer, max_lng integer, min_lng integer, animals array) returns TABLE (num bigint, month smallint, label varchar, latitude float, longitude float) as $$

declare new_string text;
declare index int;
declare arr_len int;

begin
return query select count(photos.label) as num, photos.month, photos.label, photos.latitude, photos.longitude from photos
where photos.latitude > min_lat and photos.latitude < max_lat and photos.longitude > min_lng and photos.longitude < max_lng
group by photos.month, photos.label, photos.latitude, photos.longitude;
end;

$$ language plpgsql;

Big thanks to anyone who can help me with this

**Comment by GaryAustin1** [comment, marked as answer]
Normally you would have something like `animals text[]`.
Remember with most things on the database side it is all Postgres so you can use any reference on the web with that in your search and not limit to Supabase.
